arch/arm/core/aarch64/fatal.c

```diff
@@ -162,36 +162,17 @@ void z_arm64_fatal_error(unsigned int reason, const z_arch_esf_t *esf)
 	if (reason != K_ERR_SPURIOUS_IRQ) {
 		__asm__ volatile("mrs %0, CurrentEL" : "=r" (el));
 
-		switch (GET_EL(el)) {
-		case MODE_EL1:
+		if (GET_EL(el) != MODE_EL0) {
 			__asm__ volatile("mrs %0, esr_el1" : "=r" (esr));
```

Comment:
would recommend using `read_sysreg`/`write_sysreg` APIs for consistency, maybe a separate patch.

Reply:
Ok, I'll add a new patch on top.